File: Editorial/2176_Count_Equal_and_Divisible_Pairs_in_an_Array/Go/1_Traverse_number_pairs.go
func countPairs(nums []int, k int) int {
	n := len(nums)
	res := 0 // number of pairs meeting the requirements
	for i := 0; i < n-1; i++ {
		for j := i + 1; j < n; j++ {
			if (i*j)%k == 0 && nums[i] == nums[j] {
				res++
			}
		}
	}
	return res
}

File: Editorial/2101-2200/2176_Count_Equal_and_Divisible_Pairs_in_an_Array/TypeScript/1_Traverse_number_pairs.ts
function countPairs(nums: number[], k: number): number {
    let n = nums.length;
    let res = 0; // number of pairs meeting the requirements
    for (let i = 0; i < n - 1; ++i) {
        for (let j = i + 1; j < n; ++j) {
            if ((i * j) % k === 0 && nums[i] === nums[j]) {
                ++res;
            }
        }
    }
    return res;
}
